src/shed/__init__.py

```diff
@@ -59,6 +142,8 @@ def shed(
     """Process the source code of a single module."""
     assert isinstance(source_code, str)
     assert isinstance(refactor, bool)
+    # TODO: I guess this required frozenset because of isort compatibility
+    # but we can probably relax that to an iterable[str]
     assert isinstance(first_party_imports, frozenset)
```

Comment:
I think it was more because I prefer strict validation unless there's a specific reason not to, and there's the '`str` is an iterable of `str`' footgun.

Reply:
ugh, right. I suppose I could change it to `Iterable[str]` and `assert not isinstance(first_party_imports, str)`?

Reply:
and then convert it to a frozenset? Sounds good.

Reply:
It was previously passed to `isort` as a frozenset, but now we only use it as a parameter to ruff - where I currently do (the equivalent of) `str(list(first_party_imports))` for formatting it. So if we'd only accept a `list[str]` we don't need to convert the type... not that it really matters though.